org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/preferences/OptionsConfigurationBlock.java

```diff
@@ -581,6 +584,16 @@ private void logError(String text, IOException e) {
 		uiPlugin.getLog().log(status);
 	}
 
+	public Map<String, ValueDifference<String>> getBuildAffectingPreferenceChanges() {
+		Map<String, ValueDifference<String>> changes = getPreferenceChanges();
+		Map<String, ValueDifference<String>> buildAffectingChanges = Maps.filterEntries(changes, getAffectsBuilderPredicate());
+		return buildAffectingChanges;
+	}
+
+	protected Predicate<Entry<String, ValueDifference<String>>> getAffectsBuilderPredicate() {

```

Comment:
OptionsConfigurationBlock is api so i wonder if we should make this method api too

Reply:
Should be make this a method `hasBuildAffectingPreferenceChanges()` that delegates to a utility?